modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportDeleteByQueryAction.java

```diff
@@ -0,0 +1,99 @@
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.mapper.internal.ParentFieldMapper;
+import org.elasticsearch.index.mapper.internal.RoutingFieldMapper;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+public class TransportDeleteByQueryAction extends HandledTransportAction<DeleteByQueryRequest, BulkIndexByScrollResponse> {
+
+    private final Client client;
+    private final ClusterService clusterService;
+
+    @Inject
+    public TransportDeleteByQueryAction(Settings settings, ThreadPool threadPool, ActionFilters actionFilters,
+                                        IndexNameExpressionResolver resolver, Client client,
+                                        TransportService transportService, ClusterService clusterService) {
+        super(settings, DeleteByQueryAction.NAME, threadPool, transportService, actionFilters, resolver, DeleteByQueryRequest::new);
+        this.client = client;
+        this.clusterService = clusterService;
+    }
+
+    @Override
+    protected void doExecute(Task task, DeleteByQueryRequest request, ActionListener<BulkIndexByScrollResponse> listener) {
+        ParentTaskAssigningClient client = new ParentTaskAssigningClient(this.client, clusterService.localNode(), task);
+        new AsyncDeleteBySearchAction((BulkByScrollTask) task, logger, client, threadPool, request, listener).start();
+    }
+
+    @Override
+    protected void doExecute(DeleteByQueryRequest request, ActionListener<BulkIndexByScrollResponse> listener) {
+        throw new UnsupportedOperationException("task required");
+    }
+
+    /**
+     * Simple implementation of delete-by-query using scrolling and bulk.
+     */
+    static class AsyncDeleteBySearchAction extends AbstractAsyncBulkIndexByScrollAction<DeleteByQueryRequest> {
+
+        public AsyncDeleteBySearchAction(BulkByScrollTask task, ESLogger logger, ParentTaskAssigningClient client,
+                                         ThreadPool threadPool, DeleteByQueryRequest request,
+                                         ActionListener<BulkIndexByScrollResponse> listener) {
+            super(task, logger, client, threadPool, request, request.getSearchRequest(), listener);
+        }
+
+        @Override
+        protected boolean accept(SearchHit doc) {
+            return true;
+        }
+
+        @Override
+        protected DeleteRequest buildRequest(SearchHit doc) {
+            DeleteRequest delete = new DeleteRequest();
+            delete.index(doc.index());
+            delete.type(doc.type());
+            delete.id(doc.id());
+            delete.version(doc.version());
+            return delete;
+        }
+
+        @Override
+        protected ActionRequest<?> copyMetadata(ActionRequest<?> request, SearchHit doc) {
+            copyParent(request, fieldValue(doc, ParentFieldMapper.NAME));
```

Comment:
Maybe also good to comment on how the superclass is much more update oriented and you override to skip all of that.